### DISCUSSION TITLE
Is frequency_penalty and presence_penalty working?

### DISCUSSION THREAD

**Discussion by MusicStudioNYC** [tectalichq/public-openai-client-php#9] [General] [answered]
I've been getting repetitive answers, so I cranked both penalties up to the highest (2) and did not observe any improvement. Any ideas what may be happening? Can anyone else test if the penalty parameters are working for them?

**Comment by thejamescollins** [comment, marked as answer]
Hi there @UpTheDesign,

Looking at the [Completions documentation](https://beta.openai.com/docs/api-reference/completions/create), it sounds like you are setting both `presence_penalty` and `frequency_penalty` to a value of 2, is that correct?

If you use the [OpenAI Playground](https://beta.openai.com/playground) and do the same there, are the results similar?

i.e. are the results in the Playground more or less random than your API request results?

James
